zenoh/src/api/session.rs

```diff
@@ -3170,51 +3170,46 @@ where
     OpenBuilder::new(config)
 }
 
+impl std::fmt::Display for SessionInner {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}:{}", self.runtime, self.id)
+    }
+}
+
 #[async_trait]
 impl Closee for Arc<SessionInner> {
     async fn close_inner(&self) {
+        info!(zid = %self.zid(), "close session");
+
         let Some(primitives) = zwrite!(self.state).primitives.take() else {
+            debug!(zid = %self.zid(), "session already closed");
             return;
         };
 
+        self.task_controller.terminate_all_async().await;
+
         if self.owns_runtime {
-            info!(zid = %self.zid(), "close session");
-            self.task_controller.terminate_all_async().await;
-            self.runtime.get_closee().close_inner().await;
+            debug!(zid = %self.zid(), "closing owned runtime {}", self.runtime);
```

Comment:
Isn't the runtime display just printing the zid ? If so isn't the zid = thing just redundant ?